client/oci.go

```diff
@@ -621,25 +622,30 @@ func (r *ociRegistry) getImageConfig(ctx context.Context, creds credentials, nam
 
 var errOCIDownloadNotSupported = errors.New("not supported")
 
-func (c *Client) newOCIRegistry(ctx context.Context, name string, accessTypes []accessType) (*ociRegistry, *bearerTokenCredentials, error) {
+func (c *Client) newOCIRegistry(ctx context.Context, name string, accessTypes []accessType) (*ociRegistry, *bearerTokenCredentials, string, error) {
 	// Attempt to obtain (direct) OCI registry auth token
-	registryURI, creds, err := c.ociRegistryAuth(ctx, name, accessTypes)
+	registryURI, creds, resolvedName, err := c.ociRegistryAuth(ctx, name, accessTypes)
 	if err != nil {
-		return nil, nil, errOCIDownloadNotSupported
+		return nil, nil, "", errOCIDownloadNotSupported
 	}
 
 	// Download directly from OCI registry
 	c.Logger.Logf("Using OCI registry endpoint %v", registryURI)
+	if resolvedName != name {
```

Comment:
Do we need to deal with if an older server withtout the corresponding change doesn't report any resolved name, so that `resolvedName == ""` and that overwrites `name`?